src/chocolatey.resources/helpers/functions/Generate-BinFile.ps1

```diff
@@ -0,0 +1,94 @@
+﻿# Copyright 2011 - Present RealDimensions Software, LLC & original authors/contributors from https://github.com/chocolatey/chocolatey
+# 
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# 
+#     http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+function Generate-BinFile {
+param(
+  [string] $name,
+  [string] $path,
+  [switch] $useStart,
+  [string] $command = ''
+)
+  Write-Debug "Running 'Generate-BinFile' for $name with path:`'$path`'|`$useStart:$useStart|`$command:$command";
+
+  $packageBatchFileName = Join-Path $nugetExePath "$name.bat"
+  $packageBashFileName = Join-Path $nugetExePath "$name"
+  $packageShimFileName = Join-Path $nugetExePath "$name.exe"
+
+  if (Test-Path ($packageBatchFileName)) {Remove-Item $packageBatchFileName -force}
+  if (Test-Path ($packageBashFileName)) {Remove-Item $packageBashFileName -force}
+  $originalPath = $path
+  $path = $path.ToLower().Replace($nugetPath.ToLower(), "..\").Replace("\\","\")
+
+  $ShimGenArgs = "-o `"$packageShimFileName`" -p `"$path`" -i `"$originalPath`""
+  if ($command -ne $null -and $command -ne '') {
+    $ShimGenArgs +=" -c $command"
+  }
+
+  if ($useStart) {
+    $ShimGenArgs +=" -gui"
+  }
+
+  if ($debug) {
+    $ShimGenArgs +=" -debug"
+  }
+
+  Write-Debug "Calling $ShimGen $ShimGenArgs"

```

Comment:
So `$ShimGen` was determined in the old chocolatey.ps1, which won't have run here. Follow the same pattern as used in `Get-ChocolateyUnzip.ps1` here.